main.go

```diff
@@ -271,27 +267,27 @@ func prettyClusterStatus(actor string, st *api.GlobalPinInfo) {
 	}
 }
 
+// PinCluster pins the item with given path to cluster.
 func PinCluster(b *hb.Bot, actor, path, label string) {
 	clusterPinUnpin(b, actor, path, label, true)
 	if err := writePin(path, label); err != nil {
 		botMsg(actor, fmt.Sprintf("failed to write log entry for last pin: %s", err))
 	}
 }
 
+// UnpinCluster unpins the item with given path to cluster.
 func UnpinCluster(b *hb.Bot, actor, path string) {
 	clusterPinUnpin(b, actor, path, "", false)
 }
 
+// RecoverCluster tries to recover item with give path, if it's previous pin or
+// unpin operation was failed.
 func RecoverCluster(b *hb.Bot, actor, path string) {
 	ctx := context.Background()
-	// pick up a random cluster
-	i := rand.Intn(len(clusters))
-	cluster := clusters[i]
-	addr := clusterAddrs[i]
 
-	botMsg(actor, fmt.Sprintf("Recovering pin via %s", addr))
+	botMsg(actor, fmt.Sprintf("Recovering pin with path %s", path))
```

Comment:
same for all places where a useful message using `addr` has been turned into a less useful one

Reply:
Earlier it was printing which random client was being used. Now, there isn't one but many.